src/chef-server-ctl/plugins/reindex.rb

```diff
@@ -63,6 +65,31 @@ def do_reindex(orgs_to_reindex, options)
   enable_api if options[:disable_api]
 end
 
+# Disk space verification for running reindex 
+# It needs enough disk space to hold a second copy of the data plus whatever change in size the new index might need if the schema has changed.
+# Required Disk Space = (2 * size of ES data directory)
+def verify_disk_space
+  data_dir = running_config['private_chef']['elasticsearch']['data_dir']
+  begin
+    if Dir.exist?(data_dir)
+      data_dir_size = Du.du(data_dir)
+      free_disk_space = Statfs.new("#{data_dir}/..").free_space
+      if (2.2* data_dir_size) < free_disk_space #The minimum space should be double the existing data size
+        puts "Free space is sufficient to start Elasticsearch reindex"
+      else
+        $stderr.puts "Insufficient free space on disk to complete reindex."
+        $stderr.puts "The current elasticsearch data directory contains #{data_dir_size} KB of data but only #{free_disk_space} KB is available on disk."
+        $stderr.puts "The reindex process requires at least #{2.2*data_dir_size} KB."
+        exit 1
+      end
+    else
+      puts "Elasticserach data path does not exist, so skipping the disk space verification: #{data_dir}"
+    end
+  rescue => exception
```

Comment:
We are giving writing the exception here. Is this what you are trying to suggest me? @PrajaktaPurohit

Reply:
yeah - I was thinking it might be of help to catch the individual exception that we are throwing in du and statsvfs instead of rescuing everything.

Reply:
While I don't normally like catch-all rescues, I think it makes sense in this context. 

Almost any failure out of these checks would not be actionable by the user, so we'd end up just showing them an error that meant little - and then we'd leave them stuck there with it and no path forward.  (Types of errors I can think of: du command not found, broken permissions, disk failures preventing statfs reads). 

If the error was due to something like hard drive failure or permissions, we'll still fail later - and will still have the logged exception from here to give a hint if one is needed.

TL;DR - I think this should do what it says on the tin (or function name) and only fail if there is not enough disk space. Allowing exceptions to escape would mean that the user ends up with a stack trace and confusion.